qiskit/quantum_info/operators/predicates.py

```diff
@@ -0,0 +1,158 @@
+                 rtol=RTOL_DEFAULT,
+                 atol=ATOL_DEFAULT):
+    """Test if two arrays are equal.
+
+    The final comparison is implemented using Numpy.allclose. See its
+    documentation for additional information on tolerance parameters.
+
+    If ignore_phase is True both matrices will be multiplied by
+    exp(-1j * theta) where `theta` is the first nphase for a
+    first non-zero matrix element `|a| * exp(1j * theta)`.
+
+    Args:
+        mat1 (matrix_like): a matrix
+        mat2 (matrix_like): a matrix
+        ignore_phase (bool): ignore complex-phase differences between
+            matrices [Default: False]
+        rtol (double): the relative tolerance parameter [Default {}].
+        atol (double): the absolute tolerance parameter [Default {}].
+
+    Returns:
+        bool: True if the matrices are equal or False otherwise.
+    """.format(RTOL_DEFAULT, ATOL_DEFAULT)
+
+    if atol is None:
+        atol = ATOL_DEFAULT
+    if rtol is None:
+        rtol = RTOL_DEFAULT
+    mat1 = np.array(mat1)
+    mat2 = np.array(mat2)
+    if mat1.shape != mat2.shape:
+        return False
+    if ignore_phase:
+        # Get phase of first non-zero entry of mat1 and mat2
+        # and multiply all entries by the conjugate
+        phases1 = np.angle(mat1[mat1 != 0].ravel(order='F'))
+        if len(phases1) > 0:
+            mat1 *= np.exp(-1j * phases1[0])
+        phases2 = np.angle(mat2[mat2 != 0].ravel(order='F'))
+        if len(phases2) > 0:
+            mat2 *= np.exp(-1j * phases2[0])
+    return np.allclose(mat1, mat2, rtol=rtol, atol=atol)
+
+
+def is_square_matrix(op):
+    """Test if an array is a square matrix."""
+    mat = np.array(op)
+    if mat.ndim != 2:
+        return False
+    shape = mat.shape
+    return shape[0] == shape[1]
+
+
+def is_diagonal_matrix(op, rtol=RTOL_DEFAULT, atol=ATOL_DEFAULT):
+    """Test if an array is a diagonal matrix"""
+    if atol is None:
+        atol = ATOL_DEFAULT
+    if rtol is None:
+        rtol = RTOL_DEFAULT
+    mat = np.array(op)
+    if mat.ndim != 2:
+        return False
+    return np.allclose(mat, np.diag(np.diagonal(mat)), rtol=rtol, atol=atol)
+
+
+def is_symmetric_matrix(op, rtol=RTOL_DEFAULT, atol=ATOL_DEFAULT):
+    """Test if an array is a symmetrix matrix"""
+    if atol is None:
+        atol = ATOL_DEFAULT
+    if rtol is None:
+        rtol = RTOL_DEFAULT
+    mat = np.array(op)
+    if mat.ndim != 2:
+        return False
+    return np.allclose(mat, mat.T, rtol=rtol, atol=atol)
+
+
+def is_hermitian_matrix(op, rtol=RTOL_DEFAULT, atol=ATOL_DEFAULT):
+    """Test if an array is a Hermitian matrix"""
+    if atol is None:
+        atol = ATOL_DEFAULT
+    if rtol is None:
+        rtol = RTOL_DEFAULT
+    mat = np.array(op)
+    if mat.ndim != 2:
+        return False
+    return np.allclose(mat, np.conj(mat.T), rtol=rtol, atol=atol)
+
+
+def is_positive_semidefinite_matrix(op, rtol=RTOL_DEFAULT, atol=ATOL_DEFAULT):
```

Comment:
nit: can you make the arg of all these predicate functions `mat`? currently some are `op` and some are `mat`.